myia/abstract/amerge.py

```diff
@@ -412,8 +415,22 @@ def amerge(self, x1: AbstractClassBase, x2: AbstractClassBase, forced, bp):
 
 @ovld  # noqa: F811
 def amerge(self, x1: AbstractDict, x2: AbstractDict, forced, bp):
-    args1 = (x1.entries, x1.values)
-    args2 = (x2.entries, x2.values)
+    entries_1 = x1.entries
+    entries_2 = x2.entries
+    if isinstance(entries_1, DictDesc) and isinstance(entries_2, DictDesc):
+        # No available keys, use empty dicts.
+        entries_1 = {}
+        entries_2 = {}
```

Comment:
I think this new definition could be moved to `annotation_merge`, since ultimately `DictDesc` should only be found in annotations (I think?). Furthermore, since the annotation is always going to be the first argument, I think you can simply remove the non-covered branches.